filesystem/Upload.php

```diff
@@ -335,6 +344,11 @@ class Upload_Validator {
 	 */
 	public $allowedExtensions = array();
 
+	public function __construct() {
+		$fileSize = Config::inst()->get('Upload_Validator', 'default_max_file_size');
+		$this->setAllowedMaxFileSize($fileSize);
```

Comment:
Perhaps, rather than setting the defaults against the object, the `get` function will check the instance specific settings and then check the defaults after?